tests/utils/test_file_utils.py

```diff
@@ -364,3 +370,102 @@ def test_shutil_copytree_without_file_permissions(tmp_path):
     assert set(os.listdir(dst_dir.joinpath("subdir"))) == {"subdir-file.txt"}
     assert dst_dir.joinpath("subdir/subdir-file.txt").read_text() == "testing 123"
     assert dst_dir.joinpath("top-level-file.txt").read_text() == "hi"
+
+
+def test_get_total_size_basic(tmp_path):
+    root = str(tmp_path)
+
+    subdir = os.path.join(root, "subdir")
+    os.mkdir(subdir)
+    files = {
+        "file1.txt": b"hello world",  # 11 bytes
+        "file2.txt": b"This is mlflow testing.",  # 23 bytes
+    }
+    for name, content in files.items():
+        with open(os.path.join(root, name), "wb") as fp:
+            fp.write(content)
+    with open(os.path.join(subdir, "file3.txt"), "wb") as fp:
+        fp.write(b"One file under subdir.")  # 22 bytes
+
+    assert get_total_file_size(root) == 56
+    assert get_total_file_size(subdir) == 22
+
+    path_not_exists = os.path.join(root, "does_not_exist")
+    with pytest.raises(
+        MlflowException,
+        match="does not exist.",
+    ):
+        get_total_file_size(path_not_exists)
+
+    path_file = os.path.join(root, "file1.txt")
+    with pytest.raises(
+        MlflowException,
+        match="is not a directory.",
+    ):
+        get_total_file_size(path_file)
+
+
+@pytest.fixture
+@flaky()
+def small_qa_pipeline():
+    # The return type of this model's language head is a Dict[str, Any]
+    architecture = "csarron/mobilebert-uncased-squad-v2"
+    tokenizer = transformers.AutoTokenizer.from_pretrained(architecture, low_cpu_mem_usage=True)
+    model = transformers.MobileBertForQuestionAnswering.from_pretrained(
+        architecture, low_cpu_mem_usage=True
+    )
+    return transformers.pipeline(task="question-answering", model=model, tokenizer=tokenizer)
+
+
+def test_get_total_size_transformers(small_qa_pipeline, tmp_path):
+    model_dir = tmp_path.joinpath("model")
+    small_qa_pipeline.model.save_pretrained(save_directory=model_dir)
+    tokenizer_dir = tmp_path.joinpath("components").joinpath("tokenizer")
+    small_qa_pipeline.tokenizer.save_pretrained(tokenizer_dir)
+
+    expected_size = 0
+    for folder in [model_dir, tokenizer_dir]:
+        folder = str(folder)
+        expected_size += _calcualte_expected_size(folder)
+
+    assert get_total_file_size(str(tmp_path)) == expected_size
```

Comment:
Can we test this in `tests/transformers`?

Reply:
This is essentially a test for `get_total_file_size` but not transformers. What's the benefit of moving it there

Reply:
btw why do we need to test `get_total_file_size` with transformers?

Reply:
To test it with many different file types that our native flavors use as @BenWilson2 suggested in above [comment](https://github.com/mlflow/mlflow/pull/10110#discussion_r1370745458).

Reply:
Can we log a model and read `model_size_bytes` in `tests/transformers`?

Reply:
+1 to what @harupy is suggesting. We should put this integration-style test in transformers for the benefit of CI setup having dependencies within that suite. There are auto-use fixtures that clean the huggingface cache after downloading from the hub to prevent disk issues on the CI runners as well. We don't want to have to copy all of that logic for this otherwise very fast CI suite here.